src/selection-recommendation/recommend.py

```diff
@@ -0,0 +1,206 @@
+                all_possible_times.append(" ".join(all_times[i]))
+                all_possible_section_combs.append(all_section_combs[i])
+
+    # course has lab sections but no tutorial sections
+    elif (tutorial_times == []) and (lab_times != []):
+        all_times = list(itertools.product(lecture_times, lab_times))  # all combinations of lecture/lab
+        all_section_combs = list(itertools.product(lecture_sections, lab_sections))
+
+        for i in range(len(all_times)):
+            if time_conflicts_check.no_time_conflict(all_times[i][0], all_times[i][1]):  # check for internal time conflicts
+                all_possible_times.append(" ".join(all_times[i]))
+                all_possible_section_combs.append(all_section_combs[i])
+
+    # course has tutorial sections but no lab sections
+    elif (tutorial_times != []) and (lab_times == []):
+        all_times = list(itertools.product(lecture_times, tutorial_times))  # all combinations of lecture/tutorial
+        all_section_combs = [list(i) for i in list(itertools.product(lecture_sections, tutorial_sections))]
+
+        for i in range(len(all_times)):
+            if time_conflicts_check.no_time_conflict(all_times[i][0], all_times[i][1]):  # check for internal time conflicts
+                all_possible_times.append(" ".join(all_times[i]))
+                all_possible_section_combs.append(all_section_combs[i])
+
+    # course only has lecture sections
+    else:
+        all_possible_times = lecture_times
+        all_possible_section_combs = lecture_sections
+
+    return(all_possible_times, all_possible_section_combs)
+
+
+def create_schedule(campus, *args):
+    """
+    campus: a string - either "St. George", "Scarborough", or "Mississauga"
+    args: strings of course codes e.g. "CSC148", "COG250"
+    Returns a tuple identical in format to get_all_possible_course_times()
+
+    The output times and sections will be all the schedules comprised of a combination of sections from the input courses
+    with no time conflicts. If no such combinations exist, the return value is a tuple of two empty lists.
+    """
+    all_times = []
+    all_sections = []
+    all_course_times = []
+    all_course_sections = []
+
+    for course_code in args:
+        course_times, course_sections = get_all_possible_course_times(course_code, campus)
+        all_course_times.append(course_times)
+        all_course_sections.append(course_sections)
+
+    times_combs = list(itertools.product(*all_course_times))
+    sections_combs = list(itertools.product(*all_course_sections))
+
+    for i in range(len(times_combs)):
+        has_conflicts = False
+        time_comb = times_combs[i]
+
+        for j in range(0, len(time_comb) - 1):
+            for k in range(j + 1, len(time_comb)):
+                if not time_conflicts_check.no_time_conflict(time_comb[j], time_comb[k]):
+                    has_conflicts = True
+                    break
+            if has_conflicts:
```

Comment:
the break in line 143 will directly break the loop in line 140. Only line 146, 147 are needed afaik

Reply:
sry I didn't notice the outmost loop 0_o